app/src/main/java/net/bible/android/control/backup/BackupControl.kt

```diff
@@ -67,6 +76,55 @@ class BackupControl {
         }
     }
 
+    /** backup database to custom target (email, drive etc.)
+     */
+    fun backupDatabaseViaIntent(callingActivity: Activity) {
+        val fileName = CommonDatabaseHelper.DATABASE_NAME
+        internalDbBackupDir.mkdirs()
+        FileManager.copyFile(CommonDatabaseHelper.DATABASE_NAME, internalDbDir, internalDbBackupDir)
```

Comment:
In Android 9, Sqlite has changed to now use a .db-shm and .db-wal file on top of the .db file. More info here: https://stackoverflow.com/a/54056779/877453 -- As this answer says, a PRAGMA checkpoint must first be run so that the DB file will have all the changes. I found this out while trying to look into the db file while working on the reading plan and not finding the changes that I knew should be there.

That means, also on a restore, the existing db-shm and db-wal files would need to be removed, probably.

I haven't tested this with backing up, but I'm fairly certain this will have to be addressed for Android Pie.

Reply:
Good point. Need to add this to old-style backupDatabase too.

Reply:
Fixed